### PR TITLE
Address 7.x-ISLANDORA-1749; fix field linking with sepcial characters…

### PR DESCRIPTION
**Pull request by nhart**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1749

# What does this Pull Request do?
Allows linked terms containing special characters to search properly

# What's new?
Escapes special characters based on advanced search regex and replaces forward slashes with ~slsh~

# How should this be tested?
- enable "Escape Lucene special characters" with default regex under http://site.com/admin/islandora/search/islandora_solr/settings > Advanced search block
- enable "link to search" for a particular Solr field containing special characters under http://site.com/admin/islandora/search/islandora_solr/settings > Display fields

- Search for a particular object with said field containing special characters
Clicking on a linked term should perform a proper search and not return a 400 error.

# Additional Notes:
* Does this change require documentation to be updated? possibly?
* Does this change add any new dependencies? No
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No
* Could this change impact execution of existing code? No

# Interested parties
@Islandora/7-x-1-x-committers @DiegoPino

### CODE REVIEW

**Review by rosiel — CHANGES_REQUESTED** [review]
I can't tell if this is doing what it's supposed to. I tried with five sample strings including various slashes and backslashes. 

I'm gonna try table markdown here... 

|test string	|before this PR	|With this pr, no config changes	|regex enabled|
|---|---|---|---|
|`simple/test`	|fail	|pass	|pass|
|`simple//test	`|fail	|pass	|pass|
|`simple/\test`	|fail	|pass	|fail|
|`simple\\test`	|pass	|pass	|fail|
|`simple.\/test`	|fail	|pass	|fail|

It seemed that good things came out of this PR, except if lucene regex was enabled. 

What's supposed to happen?

**Comment by DiegoPino** [comment]
@rosiel thanks for reviewing this, just can't find the time right now. You are just great

**Comment by nhart** [comment]
@rosiel  sorry for the delay. work is busy. life is busy. I will have to revisit and verify what the original use case was.

File: includes/utilities.inc [hunk 1]
```diff
@@ -388,10 +388,19 @@ function islandora_solr_prepare_solr_results($solr_results) {
         // Add link to search.
         if (in_array($field, $link_to_search)) {
           $map_to_link = function ($original_value, $formatted_value) use ($field) {
+            if (variable_get('islandora_solr_advanced_search_block_lucene_syntax_escape', FALSE)) {
+              $lucene_regex = variable_get('islandora_solr_advanced_search_block_lucene_regex_default', ISLANDORA_SOLR_QUERY_FACET_LUCENE_ESCAPE_REGEX_DEFAULT);
+              $linked_value = islandora_solr_facet_query_escape(stripcslashes($original_value), $lucene_regex);
```
Comment (adam-vessey) [inline includes/utilities.inc:393]: The `stripcslashes()` invocation makes a mess with the backslashes... though really, not sure we want to do the regex thing in this situation: The links we generate are for searches on the exact string value, right? The "regex doodad" is more to allow more permissive escaping in advanced search, intended to allow the use of more Lucene syntax, if desired.

Comment (adam-vessey) [reply includes/utilities.inc:393]: Similarly, the regex stuff is intended to pass the queries slightly different... not wrapping things in quotes: https://github.com/Islandora/islandora_solr_search/blob/53852cc5859edaba910229a3e31d6600f28b65a3/includes/blocks.inc#L761-L767

**Review by adam-vessey — CHANGES_REQUESTED** [review]
I believe we only want the `islandora_solr_replace_slashes()` in the flow, and moving to the configurable regex for escaping this query is probably not desired.

**Comment by DiegoPino** [comment]
@nhart this one was solved via https://github.com/Islandora/islandora_solr_search/commit/66549e8d24d88a17a57a73a5ef134969d2a66bee, i guess it's an internal thing of you guys. Let me know if you want to close this one or your pull actually does something different. Thanks!

**Pull request closed without merging**